src/Build/Construction/Solution/ProjectInSolution.cs

```diff
@@ -152,17 +152,20 @@ internal set
                 // Avoid loading System.Runtime.InteropServices.RuntimeInformation in full-framework
                 // cases. It caused https://github.com/NuGet/Home/issues/6918.
                 _relativePath = value;
+
+                AbsolutePath = Path.GetFullPath(Path.Combine(ParentSolution.SolutionFileDirectory, _relativePath));
 #else
-                _relativePath = FileUtilities.MaybeAdjustFilePath(value,
-                                                    baseDirectory:ParentSolution.SolutionFileDirectory ?? String.Empty);
+                _relativePath = FileUtilities.MaybeAdjustFilePath(value, ParentSolution.SolutionFileDirectory);
+
+                AbsolutePath = FileUtilities.NormalizePath(Path.Combine(ParentSolution.SolutionFileDirectory, _relativePath));
```

Comment:
Yeah I might as well.  Especially if it throws an exception for whatever reason it would be confusing to be accessing the RelativePath property and get an exception about AbsolutePath